Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs

```
@@ -62,87 +64,113 @@ void OnDestroy()
+
+                m_GameNetPortal.PlayerName = m_LocalUser.DisplayName;
+
+                switch (m_LocalLobby.OnlineMode)
+                {
+                    case OnlineMode.IpHost:
+                        Debug.Log($"Created lobby with ID: {m_LocalLobby.LobbyID} and code {m_LocalLobby.LobbyCode}, at IP:Port {m_LocalLobby.Data.IP}:{m_LocalLobby.Data.Port}");
+                        m_GameNetPortal.StartHost(m_LocalLobby.Data.IP, m_LocalLobby.Data.Port);
+                        break;
+
+                    case OnlineMode.UnityRelay:
+                        Debug.Log($"Created lobby with ID: {m_LocalLobby.LobbyID} and code {m_LocalLobby.LobbyCode}, Internal Relay Join Code{m_LocalLobby.RelayJoinCode}");
+                        m_GameNetPortal.StartUnityRelayHost();
+                        break;
+                }
+            }
+            else
+            {
+                UnblockUIAfterLoadingIsComplete();
+            }
         }
 
-        public void QueryLobbiesRequest(bool blockUI)
+        public async void QueryLobbiesRequest(bool blockUI)
         {
-            if (!AuthenticationService.Instance.IsAuthorized)
+            if (UnityServices.State != ServicesInitializationState.Initialized)
             {
                 return;
             }
 
-            m_LobbyServiceFacade.RetrieveLobbyListAsync(
-                OnSuccess,
-                OnFailure
-            );
-
-            if (blockUI)
+            try
             {
-                BlockUIWhileLoadingIsInProgress();
-            }
+                if (blockUI)
+                {
+                    BlockUIWhileLoadingIsInProgress();
+                }
 
-            void OnSuccess(QueryResponse qr)
-            {
+                await m_LobbyServiceFacade.RetrieveAndPublishLobbyListAsync();
                 UnblockUIAfterLoadingIsComplete();
             }
-
-            void OnFailure()
+            catch (Exception)
```

we can use try finally instead here

---

or just remove catch all Exception

---

and let this crash if there's an unexpected exception